### PR TITLE
Adds Vault support.

### PR DESCRIPTION
**Pull request by lordofthejars**

### CODE REVIEW

File: arquillian-ape-nosql/vault-ftest/src/main/java/org/arquillian/ape/nosql/vault/Secrets.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import com.bettercloud.vault.VaultException;
+import java.util.Map;
+
+public class Secrets {
+
+    private com.bettercloud.vault.VaultConfig vaultConfig;
+    private Vault vault;
+
+    public Secrets(String host, int port, String token) throws VaultException {
+        this.vaultConfig = new VaultConfig(String.format("http://%s:%s", host, port), token);
```
Comment (bartoszmajsak) [inline arquillian-ape-nosql/vault-ftest/src/main/java/org/arquillian/ape/nosql/vault/Secrets.java]: Second placeholder for in the format should be `%d` I believe.

File: arquillian-ape-nosql/vault-ftest/src/test/java/org/arquillian/ape/nosql/vault/VaultTest.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        final Secrets secrets = new Secrets(vault.getIpAddress(), vault.getBindPort(8200), ROOT_TOKEN);
+        final Map<String, String> data = secrets.getFooSecret();
+
+        // then
+
+        assertThat(data).containsEntry("zip", "zap").containsEntry("a", "b");
+
+    }
+
+
```
Comment (bartoszmajsak) [inline arquillian-ape-nosql/vault-ftest/src/test/java/org/arquillian/ape/nosql/vault/VaultTest.java]: I think we can save on empty lines here :)

File: arquillian-ape-nosql/vault-ftest/src/main/java/org/arquillian/ape/nosql/vault/Secrets.java (hunk 1 above)
Comment (lordofthejars) [inline arquillian-ape-nosql/vault-ftest/src/main/java/org/arquillian/ape/nosql/vault/Secrets.java]: It is not necessary, you can do `%d` or `%s` and calls the `toString` method.

Comment (bartoszmajsak) [inline arquillian-ape-nosql/vault-ftest/src/main/java/org/arquillian/ape/nosql/vault/Secrets.java]: I know, but just for correctness of what you really expect.

**Pull request merged**